docs/_docs/02_features/debugging.md

```diff
@@ -136,3 +136,158 @@ in dependency and local variable resolution.
 https://github.com/gruntwork-io/terragrunt/blob/eb692a83bee285b0baaaf4b271c66230f99b6358/docs/_docs/02_features/debugging.md
 for thoughts on other potential features to implement.
 -->
+
+### OpenTelemetry integration
+
+Terragrunt can be configured to emit telemetry in [OpenTelemetry](https://opentelemetry.io/) format, traces and metrics.
+
+Concepts:
+* [OpenTelemetry](https://opentelemetry.io/)
+* [Traces](https://opentelemetry.io/docs/concepts/signals/traces/)
+* [Metrics](https://opentelemetry.io/docs/concepts/signals/metrics/)
+* [Jaeger](https://www.jaegertracing.io/)
+
+Tracing configuration:
+* `TERRAGRUNT_TELEMETRY_TRACE_EXPORTER` - traces exporter type to be used. Currently supported values are:
+  * `none` - no trace exporting, default value.
+  * `console` - to export traces to console as JSON
+  * `otlpHttp` - to export traces to an OpenTelemetry collector over HTTP [otlptracehttp](https://pkg.go.dev/go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp)
+  * `otlpGrpc` - to export traces over gRPC [otlptracegrpc](https://pkg.go.dev/go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc)
+  * `http` - to export traces to a custom HTTP endpoint using [otlptracehttp](https://pkg.go.dev/go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp)
+* `TERRAGRUNT_TELEMERTY_TRACE_EXPORTER_HTTP_ENDPOINT` - in case of `http` exporter, this is the endpoint to which traces will be sent.
+* `TERRAGRUNT_TELEMERTY_TRACE_EXPORTER_INSECURE_ENDPOINT` - if set to true, the exporter will not validate the server's certificate, helpful for local traces collection.
+
+Metrics configuration:
+* `TERRAGRUNT_TELEMETRY_METRIC_EXPORTER` - metrics exporter type to be used. Currently supported values are:
+  * `none` - no metric exporting, default value.
+  * `console` - write metrics to console as JSONs.
+  * `otlpHttp` - export metrics to an OpenTelemetry collector over HTTP [otlpmetrichttp](https://pkg.go.dev/go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp)
+  * `grpcHttp` - export metrics to an OpenTelemetry collector over gRPC [otlpmetricgrpc](https://pkg.go.dev/go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc)
+* `TERRAGRUNT_TELEMERTY_METRIC_EXPORTER_INSECURE_ENDPOINT` - if set to true, the exporter will not validate the server's certificate, helpful for local metrics collection.
+
+## Example configurations for trace collection
+
+Example traces collection with Jaeger:
```

Comment:
NIT: could you make each example have its own sub-heading (e.g., `###`)? That would make it a lot easy easier to scan these examples and find the one you want.